simulation/sub8_gazebo/nodes/fake_perception.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python
+from geometry_msgs.msg import Pose, Point, PoseStamped, Vector3
+from mil_misc_tools import text_effects
+from mil_msgs.srv import SetGeometry
+import numpy as np
+import rospy
+from std_srvs.srv import SetBool, SetBoolResponse
+from sub8_msgs.srv import VisionRequestResponse, VisionRequest, VisionRequest2D
+from gazebo_msgs.srv import GetModelState
+from std_msgs.msg import Header
+import os
+import rospkg
+import sys
+import yaml
+
+
+rospack = rospkg.RosPack()
+config_file = os.path.join(rospack.get_path(
+    'sub8_missions'), 'sub8', 'vision_proxies.yaml')
+f = yaml.load(open(config_file, 'r'))
+
+fprint = text_effects.FprintFactory(title="SIMULATOR").fprint
+
+
+def handle_fake_perception(extra, target_object):
+    '''
+    Calls the GetModelState service from gazebo to get the realtime position of the model targeted.
+    Provides this information to the mission.
+    @param extra The target_name passed through some missions.
+    Other missions do not pass a target_name thus its lable of extra.
+    @param target_object Is the model name of the object targeted by the mission.
+    Missions that do not pass a target_name must use this.
+    '''
+
+    now = rospy.get_rostime()
+    k = np.uint32(0)
+    if extra != '':
+        target_object = extra
+    if target_object == '':
+        fprint("NO TARGET")
+        sys.exit(0)
+    model = get_position(target_object)
```

Comment:
So you are not handling offsets required for some models? It would be nice to put in vision proxies a list offsets from the model origin to where you want fake perception to return (like the start gate, centroid instead of one corner)